tempesta_fw/ss_skb.c

```diff
@@ -744,21 +755,33 @@ ss_skb_cutoff_data(const TfwStr *hdr, int skip, int tail)
 			continue;
 		}
 		it = it_zero;
-		r = __skb_fragment(c->skb, NULL,
-				   (char *)c->ptr + skip, skip - c->len, &it);
-		if (r)
+		f_len = skip - c->len;
+		f_ptr = (char *)c->ptr + skip;
+		r = __skb_fragment(c->skb, NULL, f_ptr, f_len, &it);
+		if (r < 0)
 			return r;
+		BUG_ON(r != -f_len);
 		skip = 0;
 	}
 
 	/* Cut off the tail. */
 	if (tail) {
+		struct sk_buff *f_skb;
```

Comment:
`it` is used to return the values that we may need if `EOL` is split between fragments or SKBs. For that `it` needs to be zeroed (see below). For that reason we can't use `it->ptr` and `it->skb` as functions arguments.